plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/validation/SCTValidationJob.java

```diff
@@ -40,20 +34,18 @@
 import org.yakindu.sct.ui.editor.DiagramActivator;
 
 import com.google.inject.Inject;
-import com.google.inject.Singleton;
 
 /**
  * 
  * @author andreas muelder - Initial contribution and API
  * 
  */
-@Singleton
-public class SCTValidationJob extends Job implements IMarkerType {
+public class SCTValidationJob extends Job {
 
 	@Inject
 	private IResourceValidator validator;

```

Comment:
We use Issues here to reuse parts of the Xtext infrastructure (IValidationIssueProcessor, MarkerCreator, MarkerTypeProvider, etc.) The overhead of Issue creation can be ignored since these are just simple POJOs. If it pops up in your profiler please let me know.